api/indexer.oas2.json

```diff
@@ -1427,15 +1495,37 @@
     },
     "Transaction": {
       "description": "Contains all fields common to all transactions and serves as an envelope to all transactions type.\n\nDefinition:\ndata/transactions/signedtxn.go : SignedTxn\ndata/transactions/transaction.go : Transaction\n",
+      "allOf": [
```

Comment:
I haven't seen this before. So this is taking the `TransactionBase` type and adding in the id/signature fields?

Do you need the "required" field to mark id/signature as required?

Reply:
You've got them, nice :+1:

Reply:
There is one complication when we enable searching for inner transactions.

What happens when we search for transactions and just have an inner transaction in the results?

1) if we lookup the entire tree, this works nicely because we would use this object as designed.
2) if we want to return the TransactionBase it doesn't fit so well anymore

Reply:
I agree it might look a bit confusing. I have not found better way of preserving existing `Transaction`, excluding id/sig and reusing the definition. If we decide empty id/sig is OK for inners then `TransactionBase` can go. Another option is enforce required fields outside of the spec but I do not like this approach